src/OrchardCore.Modules/OrchardCore.ReCaptcha/Users/Handlers/RegistrationFormEventHandler.cs

```diff
@@ -1,17 +1,26 @@
+using Microsoft.AspNetCore.Identity;
 using OrchardCore.ReCaptcha.Services;
+using OrchardCore.Users;
 using OrchardCore.Users.Events;
 
 namespace OrchardCore.ReCaptcha.Users.Handlers;
 
 public sealed class RegistrationFormEventHandler : RegistrationFormEventsBase
 {
     private readonly ReCaptchaService _reCaptchaService;
+    private readonly SignInManager<IUser> _signInManager;
 
-    public RegistrationFormEventHandler(ReCaptchaService reCaptchaService)
+    public RegistrationFormEventHandler(ReCaptchaService reCaptchaService, SignInManager<IUser> signInManager)
     {
         _reCaptchaService = reCaptchaService;
+        _signInManager = signInManager;
     }
 
-    public override Task RegistrationValidationAsync(Action<string, string> reportError)
-        => _reCaptchaService.ValidateCaptchaAsync(reportError);
+    public override async Task RegistrationValidationAsync(Action<string, string> reportError)
+    {
+        if (await _signInManager.GetExternalLoginInfoAsync() == null)
```

Comment:
It maybe be helpful to add a comment here explaining why not to validate when we're is an external am login.

And I would do if(... is not null)
{
    return;
}

Reply:
I don't really see the point of doing an early return when the rest of the method is a single statement, since then it just makes it more verbose. But I changed these.